--- ser2net-2.10.0/configure.ac.orig	2005-12-30 00:51:17.000000000 +0100
+++ ser2net-2.10.0/configure.ac	2006-02-06 13:42:07.000000000 +0100
@@ -16,7 +16,19 @@
 if test "$tcp_wrappers" != "no"
 then
   AC_HAVE_HEADERS(tcpd.h)
-  AC_CHECK_LIB(wrap,main)
+  ac_save_LIBS="$LIBS"
+  LIBS="-lwrap $LIBS"
+  AC_TRY_LINK([
+  #include <syslog.h>
+  #include <tcpd.h>
+  int allow_severity = LOG_INFO;
+  int deny_severity = LOG_WARNING;
+  ],[],[ac_Lib=yes],[ac_Lib=no])
+  LIBS="$ac_save_LIBS"
+  if test "$ac_Lib" = "yes" ; then
+  	AC_DEFINE_UNQUOTED(HAVE_LIBWRAP)
+  	LIBS="$LIBS -lwrap"
+  fi
 fi
 
 AC_ARG_WITH(sysfs-led-support,
@@ -104,5 +104,5 @@
 
 AC_CONFIG_MACRO_DIR([m4])
 AC_STDC_HEADERS
-AC_CHECK_LIB(nsl,main)
+AC_SEARCH_LIBS(gethostbyname, nsl)
 AC_OUTPUT([Makefile ser2net.spec])
